### ISSUE TITLE
coq_makefile: unexpected behavior of DESTDIR optional variable

### ISSUE THREAD

**Issue by CohenCyril** [coq/coq#13918]
#### Description of the problem
I get a strange result if I do:
```bash
# for nix users you can replicate this script by prefixing it with `nix-shell -p coq`
cd $(mktemp -d)
echo "Definition silly : nat := 0." > silly.v
echo "silly.v" > _CoqProject
echo "-R . silly" >> _CoqProject
coq_makefile -f _CoqProject -o Makefile
make && make DESTDIR=$(mktemp -d) install
```

Indeed `make install` will install `.vo` files in a target path obtained by concatenating the provided $DESTDIR directory with the **absolute** path of the main coq user-contrib directory...
Looking at `Makefile.conf` it appears indeed that `COQMF_COQLIB` is initialized to an absolute path rather than a path suffix. But reading the contents of `Makefile`, it performs the concatenation with `$DESTDIR` which makes no sense if `COQMF_COQLIB` is already an absolute path...

(The same issue arises with `DSTROOT`)

Since I provided no options at all except for `DESTDIR`, this looks like a bug to me.

#### Coq Version

Coq 8.8 -> 8.13

(Coq 8.7 behaves "better", the suffix is the relative path `user-contrib` instead of)

**Comment by CohenCyril** [comment]
CC @gares @vbgl @Zimmi48

**Comment by silene** [comment]
We want to concatenate the installation directory with `DESTDIR`, even if the former is an absolute path. I mean, that is the main point of `DESTDIR`: the ability to prepend it to stuff like `/usr/bin/` or `/usr/lib` so that an archive can be built with installable paths.

That said, I agree with you that it does not make sense here. But I don't know what the behavior should be in your case. Where do you think the files should be installed to?

**Comment by gares** [comment]
The peculiarity of nix is that the `-prefix` used to compile Coq is not the one you would want to be used to install any other package.
I don't think Coq makefile let's you do that today (using any documented interface).
- he wants: `DESTDIR + PREFIX + LIB`
- we have: `DESTDIR + COQLIB` (where `COQLIB = PREFIX + LIB`)

I have have misguided @CohenCyril about DESTDIR, I did not see the nix peculiarity here

**Comment by gares** [comment]
FTR paths are then put in the scope of Coq via env variables

**Comment by CohenCyril** [comment]
The problem is that I do not see where one could control the `PREFIX` variable of @gares.
Most Coq packages do not have a `./configure` and the call to `coq_makefile` is usually done within a wrapping `Makefile`, and there is very little control over it's invocation, and I wouldn't even know which variable to set to make sure `PREFIX=""`
In the doc there is `COQLIBINSTALL`, `COQDOCINSTALL` but no equivalent for `bin`, so it's not complete either...
So far the "best combination" I found is `DESTDIR=$out` and  `COQMF_COQLIB=lib/coq/`, but the latter is undocumented...

**Comment by gares** [comment]
Yes, I agree there is no support for what nix needs. I guess we can add it, in the meanwhile hacking the COQMF variable is the best option I guess.
